jax/scipy/optimize/bfgs_minimize.py

```diff
@@ -0,0 +1,128 @@
+"""The Broyden-Fletcher-Goldfarb-Shanno minimization algorithm.
+https://pages.mtu.edu/~struther/Courses/OLD/Sp2013/5630/Jorge_Nocedal_Numerical_optimization_267490.pdf
+"""
+
+import jax
+import jax.numpy as jnp
+from jax.lax import while_loop
+from .line_search import line_search
+from typing import NamedTuple, Optional
+
+BFGSResults = NamedTuple(
```

Comment:
The class interface for defining a `typing.NamedTuple` is much nicer, e.g.,
```python
class BFGSResults(NamedTuple):
  """Results from BFGS.

  Attributes:
    converged: True if minimization converges
    ...
  """
  converged: bool
  failed: bool
  k: int
  ...
```
or if you don't want to write a docstring:
```python
class BFGSResults(NamedTuple):
  converged: bool  # True if minimization converges
  failed: bool  # True if line search fails
  ...
```

Reply:
Done.